### PR TITLE
feat: added CpTimeline widget

### PR DESCRIPTION
**Pull request by justinenerio**
## Changes

- Added CpTimeline widget 
- Added storybook example

## Related issues

[Fixes: #441 (comment)](https://github.com/cryptoplease/cryptoplease-dart/pull/441#discussion_r976875407)

## Screenshots
Screen Shot 2022-09-27 at 5 39 41 PM

Screen Shot 2022-09-27 at 5 39 51 PM

## Checklist

- [x] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [x] Screenshots/video added.
- [ ] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

**Review by ookami-kb — CHANGES_REQUESTED** [review]
There are some weird gaps between tiles:

image

Also, regarding the widget API – it should be more use-case oriented. Something like this:

```dart
class CpTimelineItem {
  CpTimelineItem({
    required this.title,
    required this.subtitle,
    this.trailing,
  });

  final String title;
  final String subtitle;
  final String? trailing;
}

enum CpTimelineStatus { inProgress, success, failure }

class CpTimeline extends StatelessWidget {
  const CpTimeline({
    super.key,
    required this.items,
    required this.status,
    required this.lastItem,
  });

  final List<CpTimelineItem> items;
  final CpTimelineStatus status;
  final int lastItem;

  @override
  Widget build(BuildContext context) {
    throw UnimplementedError();
  }
}
```

Everything else widget should be able to figure out on its own.

File: packages/cryptoplease_ui/lib/src/timeline_widget.dart [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  final String? trailing;
+}
+
+enum CpTimelineStatus { inProgress, success, failure }
+
+class CpTimeline extends StatelessWidget {
+  const CpTimeline({
+    super.key,
+    required this.items,
+    required this.status,
```
Comment (ookami-kb) [inline packages/cryptoplease_ui/lib/src/timeline_widget.dart:22]: What about `lastItem` parameter? Right now it's not possible to set which item is "active".

**Review by ookami-kb — CHANGES_REQUESTED** [review]

**Comment by justinenerio** [comment]
@ookami-kb i've fixed the weird gap between tiles. also reduced padding between lines, and added the white progress bar on in progress status

**Comment by ookami-kb** [comment]
@justinenerio I've pushed some changes:

- Added `active` parameter. I remember we were discussing that it's not required for our specific use case, but it's easy to add and it feels more natural from the client's point of view.
- Extracted build methods into separate widgets.
- Added asserts for items list to be non-empty – I think it should be the client's responsibility to provide an empty widget.

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**